api/v1alpha1/stardogrole_types.go

```diff
@@ -27,22 +27,28 @@ import (
 type StardogRoleSpec struct {
 	// RoleName describes (overrides) the name of a role that will be maintained in a Stardog instance.
 	// Defaults to .metadata.name.
+	// +kubebuilder:validation:Optional
 	RoleName string `json:"roleName,omitempty"`
 	// StardogInstanceRef references the StardogInstance object in which the role is maintained.
+	// +kubebuilder:validation:Required
 	StardogInstanceRef string `json:"stardogInstanceRef,omitempty"`
 	// Permissions lists the permissions assigned to a role
+	// +kubebuilder:validation:Optional
 	Permissions []StardogPermissionSpec `json:"permissions,omitempty"`
 }
 
 // StardogPermissionSpec defines a Stardog permission assigned to a Role
 type StardogPermissionSpec struct {
-	// Action describes the action a specific permission is assigned to
+	// Action describes the action a specific permission is assigned to	
 	// +kubebuilder:validation:Enum=ALL;CREATE;DELETE;READ;WRITE;GRANT;REVOKE;EXECUTE
+	// +kubebuilder:validation:Optional
```

Comment:
Does this mean that it's OK for a permission not to have an associated action? Why would that be allowed?

Reply:
I will correct that